### PR TITLE
[extractor/mediaset] Better DRM/Paywall message + _EMBED_REGEX

### PR DESCRIPTION
**Pull request by nixxo**
### Description of your *pull request* and other information

</details>

- assumed that when no formats are found is because the content is behind paywall and DRM
- removed _extract_from_webpage as no longer needed
- cleaned up unnecessary tests and fixed/updated not working ones
- f-strings
- added _EMBED_REGEX and _TEST_WEBPAGES

reported various times by @Heros99
https://github.com/yt-dlp/yt-dlp/issues/4125#issuecomment-1322060176
https://github.com/yt-dlp/yt-dlp/pull/1463#issuecomment-1326027800
https://github.com/yt-dlp/yt-dlp/pull/2158#issuecomment-1271146511

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)

### CODE REVIEW

**Comment by Heros99** [comment]
thank you very much... i'll try it now

**Comment by bashonly** [comment]
> removed _extract_from_webpage as no longer needed

Are you sure? The generic extractor calls this to detect & extract embeds.

File: yt_dlp/extractor/mediaset.py [hunk 1]
```diff
@@ -129,71 +85,23 @@ class MediasetIE(ThePlatformBaseIE):
...
-
-            video_id = embed_qs.get('id', [None])[0]
-            if not video_id:
-                continue
-            urlh = self._request_webpage(embed_url, video_id, note='Following embed URL redirect')
-            embed_url = urlh.geturl()
-            program_guid = _program_guid(parse_qs(embed_url))
-            if program_guid:
-                yield self.url_result(embed_url)
-
```
Comment (pukkandan) [inline yt_dlp/extractor/mediaset.py:196]: Used by GenericIE for embed detection

https://github.com/yt-dlp/yt-dlp/blob/88f20eff1b13f70d332ef97fc1be175271bec130/yt_dlp/extractor/generic.py#L1550-L1562

File: yt_dlp/extractor/mediaset.py [hunk 2]
```diff
@@ -217,23 +125,25 @@ def _check_drm_formats(self, tp_formats, video_id):
...
-                    update_url_query('http://link.theplatform.%s/s/%s' % (self._TP_TLD, tp_path), {
+                    update_url_query(f'http://link.theplatform.{self._TP_TLD}/s/{tp_path}', {
                         'mbr': 'true',
                         'formats': f,
                         'assetTypes': asset_type,
-                    }), guid, 'Downloading %s SMIL data' % (f.split('+')[0]))
+                    }), guid, f'Downloading {f.split("+")[0]} SMIL data')
             except ExtractorError as e:
+                if e.orig_msg == 'None of the available releases match the specified AssetType, ProtectionScheme, and/or Format preferences':
+                    e.orig_msg = 'Content behind paywall and DRM'
```
Comment (pukkandan) [inline yt_dlp/extractor/mediaset.py:146]:
```suggestion
                    e.orig_msg = 'This video is DRM protected'
```
is the standard message used

File: yt_dlp/extractor/mediaset.py [hunk 3]
```diff
@@ -29,7 +28,7 @@ class MediasetIE(ThePlatformBaseIE):
                     '''
     _TESTS = [{
         # full episode
-        'url': 'https://www.mediasetplay.mediaset.it/video/mrwronglezionidamore/episodio-1_F310575103000102',
+        'url': 'https://mediasetinfinity.mediaset.it/video/mrwronglezionidamore/episodio-1_F310575103000102',
```
Comment (pukkandan) [inline yt_dlp/extractor/mediaset.py:31]: Leave one of the tests in old domain

File: yt_dlp/extractor/mediaset.py [hunk 4]
```diff
@@ -299,23 +209,23 @@ class MediasetShowIE(MediasetIE):  # XXX: Do not subclass from concrete IE
                     '''
     _TESTS = [{
         # TV Show webpage (general webpage)
-        'url': 'https://www.mediasetplay.mediaset.it/programmi-tv/leiene/leiene_SE000000000061',
+        'url': 'https://mediasetinfinity.mediaset.it/programmi-tv/leiene/leiene_SE000000000061',
```
Comment (pukkandan) [inline yt_dlp/extractor/mediaset.py:212]: same

File: yt_dlp/extractor/mediaset.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/mediaset.py:196]: Since we are cleaning up this extractor, could u also move the above test into `_WEBPAGE_TESTS` under the IE

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/mediaset.py (hunk 3 above)
Comment (nixxo) [reply yt_dlp/extractor/mediaset.py:31]: can I make a duplicate and make it "only_matching"?

File: yt_dlp/extractor/mediaset.py (hunk 1 above)
Comment (nixxo) [reply yt_dlp/extractor/mediaset.py:196]: added `_EMBED_REGEX` and moved `_WEBPAGE_TESTS` from generic to the mediaset extractor.

How do I test this?

Comment (bashonly) [reply yt_dlp/extractor/mediaset.py:196]: `python3 test/test_download.py TestDownload.test_Mediaset_webpage`
`python3 test/test_download.py TestDownload.test_Mediaset_webpage_1` etc

Comment (nixxo) [reply yt_dlp/extractor/mediaset.py:196]: yes, thanks. I figured it out by searching in the code how the tests are generated ;-)

**Pull request merged**